pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/event/FileScanResultReceiver.java

```diff
@@ -98,17 +98,24 @@ public void receiveMessage(PubsubMessage message, AckReplyConsumer reply) {
         }
     }
 
-    private String parseFileUploadGuid(String fileName) {
-        // For authorized uploads, the base file name should be the upload guid.
-        return Path.of(fileName).getFileName().toString();
+    private String parseFileUploadGuid(String blobName) {
+        // For authorized uploads, the base file name should start with the upload guid.
+        final var fileName = Path.of(blobName).getFileName().toString();
+        if (!fileName.contains("_")) {
+            log.error("The blob name {} doesn't have any underscores in it. It must have at least one", blobName);
+            return null;
```

Comment:
What are your thoughts on throwing a `DDPException` here? This should likely be considered a fatal error at this point, and the `null` going into the `findAndLockByGuid()` method is going to result in a SQL error instead of a clear indicator as to what's going on.